war/pom.xml

```diff
@@ -575,5 +639,124 @@ THE SOFTWARE.
             </plugins>
         </build>
     </profile>
-  </profiles>
+    <profile>
+        <id>gulp-execution</id>
+        <activation>
+            <file>
+                <exists>gulpfile.js</exists>
+            </file>
+        </activation>
+        <build>
+            <plugins>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-enforcer-plugin</artifactId>
+                    <version>1.3.1</version>
+                    <executions>
+                        <execution>
+                            <id>enforce-versions</id>
+                            <goals>
+                                <goal>enforce</goal>
+                            </goals>
+                            <configuration>
+                                <rules>
+                                    <requireMavenVersion>
+                                        <version>3.1.0</version>
+                                    </requireMavenVersion>
+                                </rules>
+                            </configuration>
+                        </execution>
+                    </executions>
+                </plugin>
+                <plugin>
+                    <groupId>com.github.eirslett</groupId>
+                    <artifactId>frontend-maven-plugin</artifactId>
+                    <version>0.0.23</version>
+
+                    <executions>
+
+                        <execution>
+                            <phase>initialize</phase>
+                            <id>install node and npm</id>
+                            <goals>
+                                <goal>install-node-and-npm</goal>
+                            </goals>
+                            <configuration>
+                                <nodeVersion>v${node.version}</nodeVersion>
+                                <npmVersion>${npm.version}</npmVersion>
+                                <!-- Use the pre-download node and npm packages. See download-maven-plugin config above. -->
+                                <downloadRoot>file:${project.build.directory}/frontend/</downloadRoot>
```

Comment:
Constructing a `file`-protocol URL in this way is unsafe. (Consider a checkout of Jenkins in a directory `\\workserver\share\My 100% kosher directory name #1\`.) Instead use `${project.baseUri}target/frontend/`.